### PR TITLE
Update bufr codes to handle one forecast at a time

### PR DESCRIPTION
**Pull request by BoCui-NOAA**
# Description

   This PR includes the follow changes to bufr sounding codes:  

   1. added the function to judge and process 3D soil variables, which are new outputs from GFSv17
   2.  modified the code to process forecast hour individually
   3. code clean-up and removal of nemsio input files that are not used anymore
   4. added a new module modpr_module.f90, which is a simplified version of sigio_module.f 
   5. removed linking with 'nemsio' and 'sigio' library in CMakeLists.txt

   With the updates of bufr codes and scripts, there is no need to add restart capability to GFS post-process job JGFS_ATMOS_POSTSND. 
  
   The related bufr job script update is another PR NOAA-EMC/global-workflow#2853

  Refs NOAA-EMC/global-workflow#1257
  Refs NOAA-EMC/global-workflow#2853
   
# Type of change

 - Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
-  Cycled test on WCOSS2
 
# Checklist
- [ ] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by aerorahul — COMMENTED** [review]
code changes look reasonable to me.
A couple of questions:
1. I notice `sigio_module` is being used.  Is it possible to remove the dependency on `sigio` from this program.  We have long deprecated the spectral dycore. I noticed only a couple places where `sigio` is referenced.
2. Since `nemsio` is removed, we can remove linking with `nemsio` library in `CMakeLists.txt`.
Thanks

**Comment by BoCui-NOAA** [comment]
>     1. I notice `sigio_module` is being used.  Is it possible to remove the dependency on `sigio` from this program.  We have long deprecated the spectral dycore. I noticed only a couple places where `sigio` is referenced.

The bufr code uses subroutine sigio_modpr(). I will read the code to see why it is called, and see if it can removed. 
> 
>     2. Since `nemsio` is removed, we can remove linking with `nemsio` library in `CMakeLists.txt`.
Will do, thanks!

**Comment by BoCui-NOAA** [comment]
Subroutine sigio_modpr is used to compute model pressures. It is sourced from: /apps/ops/prod/libs/build/v1.1.0/pkg/sigio-v2.3.2/src/sigio_module.f.  I did the following modifications to remove the dependency on 'sigio'. 

- Copied the original sigio_module.f to the folder gfs_bufr.fd and renamed it to modpr_module.f90
- Retained only the sigio_modpr subroutine and its related code
- Removed all other subroutines and associated code that were present in sigio_module.f
- Updated bufr codes to use module modpr_module instead of using and linking to 'sigio' library

File: src/gfs_bufr.fd/meteorg.f [hunk 1]
```diff
@@ -60,18 +65,15 @@ subroutine meteorg(npoint,rlat,rlon,istat,cstat,elevstn,
...
       use netcdf
-      use nemsio_module
-      use sigio_module
+      use modpr_module
       use physcons
       use mersenne_twister
 !      use funcphys, only : gfuncphys
       use funcphys
       implicit none
       include 'mpif.h'
```
Comment (aerorahul) [inline src/gfs_bufr.fd/meteorg.f]: Do we need this `include mpif.h` if this is no longer a parallel program?

**Review by aerorahul — APPROVED** [review]
looks good.
Thank you for removing the dependency on sigio library.
I have just one comment re. an `mpif.h` file that is being included.

**Comment by BoCui-NOAA** [comment]
Code meteorg.f and read_netcdf.f were updated, "mpif.h" was removed.

**Review by aerorahul — APPROVED** [review]
looks good.
thanks for accepting the suggestions and getting rid of nemsio and sigio library dependencies from this program.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by BoCui-NOAA** [comment]
Great. Thanks, Rahul and Walter.
@HuiyaChuang-NOAA

**Pull request merged**